src/crypto/variant2_int_sqrt.h

```diff
@@ -0,0 +1,75 @@
+#ifndef VARIANT2_INT_SQRT_H
+#define VARIANT2_INT_SQRT_H
+
+#include <math.h>
+#include <float.h>
+
+#define VARIANT2_INTEGER_MATH_SQRT_FIXUP(r) \
+  do { \
+    const uint64_t s = r >> 1; \
+    const uint64_t b = r & 1; \
+    const uint64_t r2 = (uint64_t)(s) * (s + b) + (r << 32) - sqrt_input; \
+    r += (((int64_t)(r2 + b) > 0) ? -1 : 0) + (((int64_t)(r2 + s + (1ULL << 32)) < 0) ? 1 : 0); \
```

Comment:
Can the `(((int64_t)(r2 + s + (1ULL << 32)) < 0) ? 1 : 0)` branch really be covered?